service/worker/deployment/deployment_client.go

```diff
@@ -240,6 +250,47 @@ func (d *DeploymentClientImpl) DescribeDeployment(ctx context.Context, namespace
 	}, nil
 }
 
+func (d *DeploymentClientImpl) GetDeploymentReachability(
+	ctx context.Context,
+	namespaceEntry *namespace.Namespace,
+	seriesName string,
+	buildID string,
+) (*workflowservice.GetDeploymentReachabilityResponse, error) {
+	deployInfo, err := d.DescribeDeployment(ctx, namespaceEntry, seriesName, buildID)
+	if err != nil {
+		return nil, err
+	}
+	currentDeploymentInfo, err := d.GetCurrentDeployment(ctx, namespaceEntry, seriesName)
+	if err != nil {
+		return nil, err
+	}
+	var currDeploymentBuildId string
+	if currentDeploymentInfo != nil && currentDeploymentInfo.GetDeployment() != nil {
+		currDeploymentBuildId = currentDeploymentInfo.GetDeployment().GetBuildId()
+	}
+
+	reachability, lastUpdateTime, err := getDeploymentReachability(
+		ctx,
+		namespaceEntry.ID().String(),
+		namespaceEntry.Name().String(),
+		seriesName,
+		buildID,
+		currDeploymentBuildId,
+		time.Now(), // approx time that currentDeployment was confirmed valid
```

Comment:
Do you need to pass this with different values in tests? if not, it can just go inside `getDeploymentReachability` itself.

Reply:
no, I just thought maybe the time would change in between the function call, but that's kinda stupid / it doesnt need to be that accurate. I'll move it inside

Reply:
moved